pkg/memtier/tracker_softdirty.go

```diff
@@ -419,11 +424,10 @@ func (t *TrackerSoftDirty) clearPageBits() {
 		}
 		if err != nil {
 			// This process cannot be tracked anymore, remove it.
```

Comment:
It should be quite rare case that we need to remove a pid that we are tracking. I think it's fine to have regionsMutex inside the loop.

Taking it outside the for loop causes a deadlock, because t.regionsPids() call acquires the same lock.

Reply:
Thank you for pointing it out. :D
Updated the PR and here is the detailed test results for the e2e tests: https://pastebin.com/raw/KdvAv4Xy

Tests summary:
PASS no-policy/n4c16/test00-routines
PASS no-policy/n4c16/test01-mover
PASS pidwatcher/n4c16/test00-pidlist
PASS pidwatcher/n4c16/test01-cgroups
PASS pidwatcher/n4c16/test02-filter-proc
PASS pidwatcher/n4c16/test03-filter-cgroups
PASS policy-age/n4c16/test00-swap-idlepage
PASS policy-age/n4c16/test01-move-softdirty
PASS policy-heat/n4c16/test00-swap-idlepage
PASS policy-heat/n4c16/test01-swap-softdirty
PASS policy-ratio/n4c16/test00-swap
PASS policy-ratio/n4c16/test01-move
FAIL tracker-damon/n4c16/test00-enable-damon